### PR TITLE
[AIRFLOW-4453] Make behavior of `none_failed` consistent with documentation (#7464)

### PR DESCRIPTION
**Pull request by ArgentFalcon**
The documentation for the `none_failed` trigger rule
(https://airflow.apache.org/docs/stable/concepts.html#trigger-rules)
describes its behavior as "all parents have not failed (`failed` or
`upstream_failed`) i.e. all parents have succeeded or been skipped."
With that definition in mind, there is no reason that the check for
`none_failed` should ever have to check for skipped upstream tasks or
set the current task to `skipped`. The current behavior causes the rule
to skip if all upstream tasks have skipped, which is more than a little
confusing. This fixes the behavior to be consistent with the documentation.

Co-authored-by: root <jcroteau@liveramp.com>

---
Issue link: WILL BE INSERTED BY [boring-cyborg](https://github.com/kaxil/boring-cyborg)

Make sure to mark the boxes below before creating PR: [x]

- [ ] Description above provides context of the change
- [ ] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [ ] Unit tests coverage for changes (not needed for documentation changes)
- [ ] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [ ] Relevant documentation is updated including usage instructions.
- [ ] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

Based on this PR:
https://github.com/lyft/etl/pull/32271

### CODE REVIEW

**Comment by ArgentFalcon** [comment]
cc @astahlman @milton0825 @shoumitras

File: airflow/ti_deps/deps/trigger_rule_dep.py [hunk 1]
```diff
@@ -211,6 +214,15 @@ def _evaluate_trigger_rule(
                     "upstream_tasks_state={2}, upstream_task_ids={3}"
                     .format(tr, num_failures, upstream_tasks_state,
                             task.upstream_task_ids))
+        elif tr == TR.NONE_FAILED_OR_SKIPPED:
+            num_failures = upstream - successes - skipped
```
Comment (milton0825) [inline airflow/ti_deps/deps/trigger_rule_dep.py:218]: So actually the name should be `all_success_or_skipped` would be more accurate?

Comment (ArgentFalcon) [reply airflow/ti_deps/deps/trigger_rule_dep.py:218]: I'm not entirely sure. I'm kind of confused how it differs from NONE_FAILED, but no one seems to have an issue with it upstream :/

Comment (milton0825) [reply airflow/ti_deps/deps/trigger_rule_dep.py:218]: Yeah the implementation looks exactly the same :/

Comment (milton0825) [reply airflow/ti_deps/deps/trigger_rule_dep.py:218]: Actually no. For `NONE_FAILED_OR_SKIPPED`, the task won't be skipped if there is at least one success

**Review by milton0825 — APPROVED** [review]

**Comment by ArgentFalcon** [comment]
💨

**Pull request merged**